From 62db172b5ca68e4e9f73311503e795f3cc2822c1 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Thu, 23 Jan 2025 16:03:29 +0100
Subject: [PATCH 1/2] Support getting rootcling_stage1 from external build

---
 core/rootcling_stage1/CMakeLists.txt | 68 ++++++++++++++++------------
 1 file changed, 39 insertions(+), 29 deletions(-)

diff --git a/core/rootcling_stage1/CMakeLists.txt b/core/rootcling_stage1/CMakeLists.txt
index f448fcb624..e46db7d9ff 100644
--- a/core/rootcling_stage1/CMakeLists.txt
+++ b/core/rootcling_stage1/CMakeLists.txt
@@ -7,39 +7,49 @@
 ############################################################################
 # CMakeLists.txt file for building ROOT core/rootcling_stage1 package
 ############################################################################
+set(ROOTCLING_STAGE1_PATH "" CACHE FILEPATH "rootcling_stage1 binary to use for cross-compilation")
 
-if(WIN32)
-  set_source_files_properties(src/rootcling_stage1.cxx PROPERTIES COMPILE_FLAGS "-D_WIN32 -DNOMINMAX")
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /ignore:4217,4286")
-else()
-  if(CXX_HAS_fno_rtti)
-    set_source_files_properties(src/rootcling_stage1.cxx PROPERTIES COMPILE_FLAGS "-fno-rtti")
+if(CMAKE_CROSSCOMPILING)
+  if (NOT ROOTCLING_STAGE1_PATH)
+    message(FATAL_ERROR "ROOTCLING_STAGE1_PATH must be set when cross-compiling")
   endif()
-endif()
 
-if(builtin_clang)
-  set(CLING_LIBRARIES "clingMetaProcessor")
+  add_executable(rootcling_stage1 IMPORTED GLOBAL)
+  set_property(TARGET rootcling_stage1 PROPERTY IMPORTED_LOCATION ${ROOTCLING_STAGE1_PATH})
 else()
-  list(APPEND CLING_LIBRARIES ${CLING_DEPEND_LIBS})
-  link_directories("${LLVM_LIBRARY_DIR}")
-endif()
+  if(WIN32)
+    set_source_files_properties(src/rootcling_stage1.cxx PROPERTIES COMPILE_FLAGS "-D_WIN32 -DNOMINMAX")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /ignore:4217,4286")
+  else()
+    if(CXX_HAS_fno_rtti)
+      set_source_files_properties(src/rootcling_stage1.cxx PROPERTIES COMPILE_FLAGS "-fno-rtti")
+    endif()
+  endif()
 
-ROOT_EXECUTABLE(rootcling_stage1 src/rootcling_stage1.cxx
-                              $<TARGET_OBJECTS:Clib>
-                              $<TARGET_OBJECTS:ClingUtils>
-                              $<TARGET_OBJECTS:Dictgen>
-                              $<TARGET_OBJECTS:Foundation_Stage1>
-                              LIBRARIES ${CLING_LIBRARIES} ${CMAKE_DL_LIBS} ${CMAKE_THREAD_LIBS_INIT} ${ROOT_ATOMIC_LIBS}
-                              NOINSTALL)
+  if(builtin_clang)
+    set(CLING_LIBRARIES "clingMetaProcessor")
+  else()
+    list(APPEND CLING_LIBRARIES ${CLING_DEPEND_LIBS})
+    link_directories("${LLVM_LIBRARY_DIR}")
+  endif()
 
-target_include_directories(rootcling_stage1 PRIVATE
-   ${CMAKE_SOURCE_DIR}/core/foundation/inc    # for RConfig.hxx
-   ${CMAKE_SOURCE_DIR}/core/dictgen/res # for rootcling_impl.h
-   ${CMAKE_BINARY_DIR}/ginclude         # for RConfigure.h
-)
-if (CMAKE_SYSTEM_NAME MATCHES FreeBSD)
-  target_link_libraries(rootcling_stage1 util procstat)
-endif()
+  ROOT_EXECUTABLE(rootcling_stage1 src/rootcling_stage1.cxx
+                                $<TARGET_OBJECTS:Clib>
+                                $<TARGET_OBJECTS:ClingUtils>
+                                $<TARGET_OBJECTS:Dictgen>
+                                $<TARGET_OBJECTS:Foundation_Stage1>
+                                LIBRARIES ${CLING_LIBRARIES} ${CMAKE_DL_LIBS} ${CMAKE_THREAD_LIBS_INIT} ${ROOT_ATOMIC_LIBS}
+                                NOINSTALL)
+
+  target_include_directories(rootcling_stage1 PRIVATE
+    ${CMAKE_SOURCE_DIR}/core/foundation/inc    # for RConfig.hxx
+    ${CMAKE_SOURCE_DIR}/core/dictgen/res # for rootcling_impl.h
+    ${CMAKE_BINARY_DIR}/ginclude         # for RConfigure.h
+  )
+  if (CMAKE_SYSTEM_NAME MATCHES FreeBSD)
+    target_link_libraries(rootcling_stage1 util procstat)
+  endif()
 
-set_target_properties(rootcling_stage1 PROPERTIES RUNTIME_OUTPUT_DIRECTORY src)
-add_dependencies(rootcling_stage1 ClingUtils)
+  set_target_properties(rootcling_stage1 PROPERTIES RUNTIME_OUTPUT_DIRECTORY src)
+  add_dependencies(rootcling_stage1 ClingUtils)
+endif()
-- 
2.47.1

