From e3e5d392d84cdade278216d270b1454c335f73b4 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Wed, 5 Mar 2025 23:05:38 +0100
Subject: [PATCH] Disable modules.idx

---
 CMakeLists.txt | 23 -----------------------
 1 file changed, 23 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9e2ccc63f6..94768c8b2c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -501,29 +501,6 @@ if(Vc_INCLUDE_DIR)
   set(MODULES_ROOT_INCPATH "ROOT_INCLUDE_PATH=${Vc_INCLUDE_DIR}:${ROOT_INCLUDE_PATH}")
 endif()
 
-# modules.idx
-if(runtime_cxxmodules)
-  ROOT_GET_LIBRARY_OUTPUT_DIR(library_output_dir)
-  get_property(modules_idx_deps GLOBAL PROPERTY modules_idx_deps_property)
-  if(WIN32)
-    set(modules_idx_cmd COMMAND ${CMAKE_COMMAND} -E env PATH="${library_output_dir}\\\;%PATH%"
-                                ROOTIGNOREPREFIX=1 ROOT_HIST=0 $<TARGET_FILE:root.exe> -l -q -b)
-  else()
-    set(modules_idx_cmd COMMAND ${ld_library_path}=${library_output_dir}:$ENV{${ld_library_path}}
-                                ROOTIGNOREPREFIX=1 ROOT_HIST=0 $<TARGET_FILE:root.exe> -l -q -b)
-  endif()
-  add_custom_command(OUTPUT ${library_output_dir}/modules.idx
-                     COMMAND ${CMAKE_COMMAND} -E remove -f modules.idx modules.timestamp
-                     ${modules_idx_cmd}
-                     WORKING_DIRECTORY ${library_output_dir}
-                     DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts
-                             ${modules_idx_deps})
-  add_custom_target(modules_idx ALL DEPENDS ${library_output_dir}/modules.idx)
-  add_dependencies(modules_idx ${modules_idx_deps})
-  set_property(TARGET modules_idx PROPERTY modules_idx_file ${library_output_dir}/modules.idx)
-  set_directory_properties(PROPERTIES ADDITIONAL_CLEAN_FILES "${library_output_dir}/modules.timestamp")
-endif()
-
 #---hsimple.root---------(use the executable for clearer dependencies and proper return code)---
 if(CMAKE_CROSSCOMPILING)
   message(WARNING "Cross-compiling: hsimple.root will not be generated")
-- 
2.45.0

