From 05991ef34e760e2dcec9fb64b3900abbb8ddaac7 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 26 Jan 2025 18:28:24 +0100
Subject: [PATCH 1/2] Enforce fail-on-missing with afterimage dependencies

---
 cmake/modules/SearchInstalledSoftware.cmake | 24 +++++++++++++++++----
 1 file changed, 20 insertions(+), 4 deletions(-)

diff --git a/cmake/modules/SearchInstalledSoftware.cmake b/cmake/modules/SearchInstalledSoftware.cmake
index 76d9eee2ce..e641a71809 100644
--- a/cmake/modules/SearchInstalledSoftware.cmake
+++ b/cmake/modules/SearchInstalledSoftware.cmake
@@ -437,15 +437,27 @@ if(asimage)
 endif()
 if(asimage)
   set(ASEXTRA_LIBRARIES)
-  find_Package(GIF)
+  if (fail-on-missing)
+    find_package(GIF REQUIRED)
+  else()
+    find_package(GIF)
+  endif()
   if(GIF_FOUND)
     set(ASEXTRA_LIBRARIES ${ASEXTRA_LIBRARIES} ${GIF_LIBRARIES})
   endif()
-  find_Package(TIFF)
+  if (fail-on-missing)
+    find_package(TIFF REQUIRED)
+  else()
+    find_package(TIFF)
+  endif()
   if(TIFF_FOUND)
     set(ASEXTRA_LIBRARIES ${ASEXTRA_LIBRARIES} ${TIFF_LIBRARIES})
   endif()
-  find_Package(PNG)
+  if (fail-on-missing)
+    find_package(PNG REQUIRED)
+  else()
+    find_package(PNG)
+  endif()
   if(PNG_FOUND)
     set(ASEXTRA_LIBRARIES ${ASEXTRA_LIBRARIES} ${PNG_LIBRARIES})
     # Some missing variables needed for external PNG build
@@ -453,7 +465,11 @@ if(asimage)
     # apparently there will be two set of includes here (needs to be selected only last that was passed: PNG_INCLUDE_DIR)
     list(GET PNG_INCLUDE_DIRS 0 PNG_INCLUDE_DIR)
   endif()
-  find_Package(JPEG)
+  if (fail-on-missing)
+    find_package(JPEG REQUIRED)
+  else()
+    find_package(JPEG)
+  endif()
   if(JPEG_FOUND)
     set(ASEXTRA_LIBRARIES ${ASEXTRA_LIBRARIES} ${JPEG_LIBRARIES})
   endif()
-- 
2.47.0

