From 7a1c86310d99e3d95ca562abbe16d56a641c23f7 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Wed, 22 Jan 2025 23:45:02 +0100
Subject: [PATCH 1/2] Support linking with mold>=2.32.0

---
 CMakeLists.txt | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2a412d5deb..f88544786c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -696,7 +696,15 @@ if(testing)
 endif()
 
 if(LLVM_LINKER_IS_MOLD)
-  message(FATAL_ERROR "The mold linker is not supported by ROOT. Please use a different linker")
+  execute_process(
+    COMMAND mold --version
+    OUTPUT_VARIABLE MOLD_VERSION
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  string(REGEX REPLACE "mold ([0-9]+\\.[0-9]+\\.[0-9]+).*" "\\1" MOLD_VERSION "${MOLD_VERSION}")
+  if(MOLD_VERSION VERSION_LESS "2.32.0")
+    message(FATAL_ERROR "The mold linker version ${MOLD_VERSION} is not supported by ROOT. Please use mold >= 2.32.0 or a different linker")
+  endif()
 endif()
 
 cmake_host_system_information(RESULT PROCESSOR QUERY PROCESSOR_DESCRIPTION)
-- 
2.47.1

