From 7703b3752b599d808cb17a360f572d91f9364e09 Mon Sep 17 00:00:00 2001
From: Enrico Guiraud <enrico.guiraud@cern.ch>
Date: Fri, 20 Aug 2021 11:05:20 +0200
Subject: [PATCH] Remove -O1 from default cling args

---
 core/metacling/src/TCling.cxx | 1 -
 1 file changed, 1 deletion(-)

diff --git a/core/metacling/src/TCling.cxx b/core/metacling/src/TCling.cxx
index 362695b035..0862898f01 100644
--- a/core/metacling/src/TCling.cxx
+++ b/core/metacling/src/TCling.cxx
@@ -1424,7 +1424,6 @@ TCling::TCling(const char *name, const char *title, const char* const argv[])
 
       clingArgsStorage.push_back("-Wno-undefined-inline");
       clingArgsStorage.push_back("-fsigned-char");
-      clingArgsStorage.push_back("-O1");
       // Disable optimized register allocation which is turned on automatically
       // by -O1, but seems to require -O2 to not explode in run time.
       clingArgsStorage.push_back("-mllvm");
-- 
2.34.1

