From 58ca7b5916c14605327f64a555e8460f283a3035 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sun, 6 Dec 2020 21:42:47 +0100
Subject: [PATCH] Hackily fix cross-compilation

---
 configure.in | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/graf2d/asimage/src/libAfterImage/configure.in b/graf2d/asimage/src/libAfterImage/configure.in
index 2a479be..678e15d 100644
--- a/graf2d/asimage/src/libAfterImage/configure.in
+++ b/graf2d/asimage/src/libAfterImage/configure.in
@@ -602,10 +602,7 @@ else
 	                   	[have_ttf=yes],AC_MSG_RESULT(no))
 	  	fi;
 	  elif test "x$have_freetype_freetype" = "xyes" ; then
-		AC_TRY_RUN([#include <ft2build.h> 
-			    #include <freetype/freetype.h>
-                            int main(){FT_Face f; return 0;}],
-	                [have_ttf=yes],AC_MSG_RESULT(no))
+		have_ttf=yes
 	  elif test "x$have_freetype" = "xyes" ; then
 		AC_TRY_RUN([#include <ft2build.h>
 			#include <freetype.h>
-- 
2.27.0

