From e8ec3aa2ad4062150a46cb7c7bfc57615bf7be9c Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Thu, 27 Jun 2019 17:59:15 +0200
Subject: [PATCH] Hack to set CMAKE_CXX_FLAGS correctly

---
 cmake/modules/CheckCompiler.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/modules/CheckCompiler.cmake b/cmake/modules/CheckCompiler.cmake
index dcf3c5aea0..ea23d1bad8 100644
--- a/cmake/modules/CheckCompiler.cmake
+++ b/cmake/modules/CheckCompiler.cmake
@@ -115,7 +115,7 @@ if(NOT CMAKE_CXX_STANDARD MATCHES "11|14|17")
 endif()
 
 # needed by roottest, to be removed once roottest is fixed
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++${CMAKE_CXX_STANDARD}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++1z")
 
 #---Check for libcxx option------------------------------------------------------------
 if(libcxx)
-- 
2.21.0

