From 308ee1dff760f440e6fb02129dab1c748b7d9220 Mon Sep 17 00:00:00 2001
From: Vincenzo Eduardo Padulano <vincenzo.eduardo.padulano@cern.ch>
Date: Wed, 19 Jun 2024 13:07:20 +0200
Subject: [PATCH] force one modulemap for darwin

---
 core/clingutils/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/clingutils/CMakeLists.txt b/core/clingutils/CMakeLists.txt
index b231dd7ba4..6906216f81 100644
--- a/core/clingutils/CMakeLists.txt
+++ b/core/clingutils/CMakeLists.txt
@@ -120,9 +120,9 @@ if (runtime_cxxmodules)
 
   # We need to override the default modulemap because instead of producing a
   # single std.pcm, produces hundreds of pcms. This changed with sdk 15.3.
-  if (CMAKE_SYSTEM_NAME MATCHES Darwin AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 15.0.0.15000309)
-    set(custom_modulemaps ${custom_modulemaps} std_darwin.modulemap)
-  endif()
+  # if (CMAKE_SYSTEM_NAME MATCHES Darwin AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 15.0.0.15000309)
+  set(custom_modulemaps ${custom_modulemaps} std_darwin.modulemap)
+  # endif()
 
   if (NOT libcxx)
     if (MSVC)
-- 
2.39.3 (Apple Git-145)

