From f98a44f1a2bdf1eccebe552fbb01e09e04c9ee2c Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Wed, 12 Feb 2025 08:40:04 +0100
Subject: [PATCH] Don't try to generate hsimple.root when cross-compiling

---
 CMakeLists.txt | 44 +++++++++++++++++++++++---------------------
 1 file changed, 23 insertions(+), 21 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index be930aea2a..9e2ccc63f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -525,29 +525,31 @@ if(runtime_cxxmodules)
 endif()
 
 #---hsimple.root---------(use the executable for clearer dependencies and proper return code)---
-add_custom_target(hsimple ALL DEPENDS tutorials/hsimple.root)
-add_dependencies(hsimple onepcm)
-if(WIN32)
-  set(hsimple_cmd COMMAND ${CMAKE_COMMAND} -E env PATH="${CMAKE_RUNTIME_OUTPUT_DIRECTORY}\\\;%PATH%"
-                          ROOTIGNOREPREFIX=1 ROOT_HIST=0 $<TARGET_FILE:root.exe> -l -q -b -n -x hsimple.C -e return)
+if(CMAKE_CROSSCOMPILING)
+  message(WARNING "Cross-compiling: hsimple.root will not be generated")
 else()
-  set(hsimple_cmd COMMAND ${MODULES_ROOT_INCPATH} ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
-                          ROOTIGNOREPREFIX=1 ROOT_HIST=0 $<TARGET_FILE:root.exe> -l -q -b -n -x hsimple.C -e return)
-endif()
-add_custom_command(OUTPUT tutorials/hsimple.root
-                   ${hsimple_cmd}
-                   WORKING_DIRECTORY tutorials
-                   DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts)
-install(FILES ${CMAKE_BINARY_DIR}/tutorials/hsimple.root DESTINATION ${CMAKE_INSTALL_TUTDIR} COMPONENT tests)
-
-if(runtime_cxxmodules)
-  add_dependencies(hsimple modules_idx)
-endif()
+  add_custom_target(hsimple ALL DEPENDS tutorials/hsimple.root)
+  add_dependencies(hsimple onepcm)
+  if(WIN32)
+    add_custom_command(OUTPUT tutorials/hsimple.root
+                      COMMAND set PATH=${CMAKE_RUNTIME_OUTPUT_DIRECTORY} &&
+                              set ROOTIGNOREPREFIX=1 && set ROOT_HIST=0 &&
+                              $<TARGET_FILE:root.exe> -l -q -b -n -x hsimple.C -e return
+                      WORKING_DIRECTORY tutorials
+                      DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts)
+  else()
+    add_custom_command(OUTPUT tutorials/hsimple.root
+                      COMMAND ${MODULES_ROOT_INCPATH} ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
+                              ROOTIGNOREPREFIX=1 ROOT_HIST=0
+                              $<TARGET_FILE:root.exe> -l -q -b -n -x hsimple.C -e return
+                      WORKING_DIRECTORY tutorials
+                      DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts)
+  endif()
+  install(FILES ${CMAKE_BINARY_DIR}/tutorials/hsimple.root DESTINATION ${CMAKE_INSTALL_TUTDIR} COMPONENT tests)
 
-#---copy special headers required for building on Windows----------------------------------------
-if(WIN32)
-  file(COPY cmake/win/w32pragma.h DESTINATION ${CMAKE_BINARY_DIR}/include/)
-  file(COPY cmake/win/sehmap.h DESTINATION ${CMAKE_BINARY_DIR}/include/)
+  if(runtime_cxxmodules)
+    add_dependencies(hsimple modules_idx)
+  endif()
 endif()
 
 #---version--------------------------------------------------------------------------------------
-- 
2.45.0

