From 0ff549745cc0c1f464b6fbc30e8c4ae29c28e5df Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Mon, 8 Feb 2021 00:23:35 +0100
Subject: [PATCH] Use find_package for llvm/clang

---
 interpreter/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/interpreter/CMakeLists.txt b/interpreter/CMakeLists.txt
index 39dc895cce..4e0556af48 100644
--- a/interpreter/CMakeLists.txt
+++ b/interpreter/CMakeLists.txt
@@ -286,6 +286,9 @@ else()
 
   set(LLVM_DIR "${LLVM_BINARY_DIR}")
 
+  find_package(LLVM REQUIRED HINTS "${LLVM_CMAKE_PATH}")
+  find_package(Clang REQUIRED HINTS "${LLVM_CMAKE_PATH}")
+
   # Normalize LLVM_CMAKE_PATH. --cmakedir might contain backslashes.
   # CMake assumes slashes as PATH.
   file(TO_CMAKE_PATH ${LLVM_CONFIG_CMAKE_PATH} LLVM_CMAKE_PATH)
@@ -449,7 +452,7 @@ if (builtin_cling)
   # LLVM doesn't really give us a API to get this with an in-source build
   # so we just use the normal way of doing this and read the llvm directory
   # compilation properties.
-  get_directory_property(LLVM_DEFS DIRECTORY llvm/src COMPILE_DEFINITIONS)
+  set(LLVM_DEFS "_GNU_SOURCE;__STDC_CONSTANT_MACROS;__STDC_FORMAT_MACROS;__STDC_LIMIT_MACROS")
   # Turns DEFINE1;DEFINE2 to -DDEFINE1 -DDEFINE2
   string (REPLACE ";" " -D" LLVM_DEFS ";${LLVM_DEFS}")
 
-- 
2.30.0

