From 3733ba086d4ec51ed81e961c4712713924c87042 Mon Sep 17 00:00:00 2001
From: Chris Burr <chrisburr@users.noreply.github.com>
Date: Tue, 2 Jun 2020 06:58:40 +0200
Subject: [PATCH 2/2] Fix DynamicLibraryManagerSymbol for external LLVM

---
 .../cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp   | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp b/interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp
index 3f028d35dc..982ea77f67 100644
--- a/interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp
+++ b/interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp
@@ -25,15 +25,9 @@
 #include <unordered_set>
 #include <vector>
 
-#include "llvm/Config/config.h" // Get configuration settings
-
-#if defined(HAVE_DLFCN_H) && defined(HAVE_DLOPEN)
 #include <dlfcn.h>
-#endif
 
-#ifdef HAVE_UNISTD_H
 #include <unistd.h>
-#endif // HAVE_UNISTD_H
 
 #ifdef __APPLE__
 #include <mach-o/dyld.h>
-- 
2.27.0

