From 4f5c55238bf7f1726811eae6beb4bd84c846876d Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Mon, 19 Apr 2021 15:23:24 +0200
Subject: [PATCH] Set LLVM_REQUIRES_EH for Cling

---
 interpreter/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/interpreter/CMakeLists.txt b/interpreter/CMakeLists.txt
index 39dc895cce..71ca52ee5c 100644
--- a/interpreter/CMakeLists.txt
+++ b/interpreter/CMakeLists.txt
@@ -323,6 +323,8 @@ else()
     "Set to ON to force using an old, unsupported host toolchain." OFF)
   option(CLANG_ENABLE_BOOTSTRAP "Generate the clang bootstrap target" OFF)
 
+  set(LLVM_REQUIRES_EH ON)
+
   include(AddLLVM)
   include(TableGen)
   include(HandleLLVMOptions)
-- 
2.27.0

