From 4e277f4dc0a1b0b65740dda1a70b1f7c2cc71d00 Mon Sep 17 00:00:00 2001
From: Vincenzo Eduardo Padulano <vincenzo.eduardo.padulano@cern.ch>
Date: Thu, 12 Dec 2024 14:56:53 +0100
Subject: [PATCH] [build] Comply with CMP0175

---
 bindings/pyroot/pythonizations/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bindings/pyroot/pythonizations/CMakeLists.txt b/bindings/pyroot/pythonizations/CMakeLists.txt
index db0d082c183ca..aa26c1773ba8c 100644
--- a/bindings/pyroot/pythonizations/CMakeLists.txt
+++ b/bindings/pyroot/pythonizations/CMakeLists.txt
@@ -192,9 +192,9 @@ endif()
 # Compile .py files
 foreach(py_source ${py_sources})
   add_custom_command(TARGET ${libname}
+                      POST_BUILD
                       COMMAND ${Python3_EXECUTABLE} -m py_compile ${localruntimedir}/${py_source}
                       COMMAND ${Python3_EXECUTABLE} -O -m py_compile ${localruntimedir}/${py_source}
-                      DEPENDS ${localruntimedir}/${py_source}
                       COMMENT "Compiling PyROOT source ${py_source} for Python ${Python3_VERSION}")
 endforeach()
 
