From 07c5467ab36492c7a8cc57814a49b7840b1271ef Mon Sep 17 00:00:00 2001
From: Vincenzo Eduardo Padulano <vincenzo.eduardo.padulano@cern.ch>
Date: Tue, 20 Jan 2026 11:42:52 +0100
Subject: [PATCH] try_run cannot be used when cross-compiling

---
 cmake/modules/RootConfiguration.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/modules/RootConfiguration.cmake b/cmake/modules/RootConfiguration.cmake
index e7e95cae4a3..8d6e3377211 100644
--- a/cmake/modules/RootConfiguration.cmake
+++ b/cmake/modules/RootConfiguration.cmake
@@ -504,7 +504,7 @@ endif()
 # The hardware interference size must be stable across all TUs in a ROOT build, so we need to save it in RConfigure.hxx
 # Since it can vary for different compilers or tune settings, we cannot base the ABI on a value that might change,
 # even be different between compiler and interpreter, or when ROOT is compiled on a different machine.
-if(CMAKE_VERSION VERSION_GREATER 3.24) # For older CMake, we simply fall back to 64
+if(CMAKE_VERSION VERSION_GREATER 3.24 AND NOT CMAKE_CROSSCOMPILING) # For older CMake, we simply fall back to 64
 set(test_interference_size "
 #include <new>
 #include <iostream>
-- 
2.52.0

