From 8532fc7f32654866d44e516905ac96e5f0c7828d Mon Sep 17 00:00:00 2001
From: Vincenzo Eduardo Padulano <vincenzo.eduardo.padulano@cern.ch>
Date: Tue, 26 Aug 2025 09:52:07 +0200
Subject: [PATCH] Disable hsimple.root generation if cross-compiling

---
 CMakeLists.txt | 43 +++++++++++++++++++++++--------------------
 1 file changed, 23 insertions(+), 20 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8fd9ebb1d39..ce1ef6ac8ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -602,29 +602,32 @@ if(runtime_cxxmodules)
 endif()
 
 #---hsimple.root---------(use the executable for clearer dependencies and proper return code)---
-add_custom_target(hsimple ALL DEPENDS tutorials/hsimple.root)
-add_dependencies(hsimple onepcm)
-if(WIN32)
-  add_custom_command(OUTPUT tutorials/hsimple.root
-                     COMMAND set PATH=${CMAKE_RUNTIME_OUTPUT_DIRECTORY} &&
-                             set ROOTIGNOREPREFIX=1 && set ROOT_HIST=0 &&
-                             $<TARGET_FILE:root.exe> -l -q -b -n -x hsimple.C -e return
-                     WORKING_DIRECTORY tutorials
-                     DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts)
+if(CMAKE_CROSSCOMPILING)
+  message(WARNING "Cross-compiling: hsimple.root will not be generated")
 else()
-  add_custom_command(OUTPUT tutorials/hsimple.root
-                     COMMAND ${MODULES_ROOT_INCPATH} ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
-                             ROOTIGNOREPREFIX=1 ROOT_HIST=0
-                             $<TARGET_FILE:root.exe> -l -q -b -n -x hsimple.C -e return
-                     WORKING_DIRECTORY tutorials
-                     DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts)
-endif()
-install(FILES ${CMAKE_BINARY_DIR}/tutorials/hsimple.root DESTINATION ${CMAKE_INSTALL_TUTDIR} COMPONENT tests)
+  add_custom_target(hsimple ALL DEPENDS tutorials/hsimple.root)
+  add_dependencies(hsimple onepcm)
+  if(WIN32)
+    add_custom_command(OUTPUT tutorials/hsimple.root
+                      COMMAND set PATH=${CMAKE_RUNTIME_OUTPUT_DIRECTORY} &&
+                              set ROOTIGNOREPREFIX=1 && set ROOT_HIST=0 &&
+                              $<TARGET_FILE:root.exe> -l -q -b -n -x hsimple.C -e return
+                      WORKING_DIRECTORY tutorials
+                      DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts)
+  else()
+    add_custom_command(OUTPUT tutorials/hsimple.root
+                      COMMAND ${MODULES_ROOT_INCPATH} ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
+                              ROOTIGNOREPREFIX=1 ROOT_HIST=0
+                              $<TARGET_FILE:root.exe> -l -q -b -n -x hsimple.C -e return
+                      WORKING_DIRECTORY tutorials
+                      DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts)
+  endif()
+  install(FILES ${CMAKE_BINARY_DIR}/tutorials/hsimple.root DESTINATION ${CMAKE_INSTALL_TUTDIR} COMPONENT tests)
 
-if(runtime_cxxmodules)
-  add_dependencies(hsimple modules_idx)
+  if(runtime_cxxmodules)
+    add_dependencies(hsimple modules_idx)
+  endif()
 endif()
-
 #---version--------------------------------------------------------------------------------------
 if(NOT WIN32)
 add_custom_target(version COMMAND ${Python3_EXECUTABLE} ${CMAKE_SOURCE_DIR}/build/unix/makeversion.py
-- 
2.51.0

