From a9f89309fb1d4443cfd7bc1881061a79f48db58f Mon Sep 17 00:00:00 2001
From: Vincenzo Eduardo Padulano <vincenzo.eduardo.padulano@cern.ch>
Date: Tue, 28 May 2024 17:02:01 +0200
Subject: [PATCH] apple-no-link-libpython

---
 cmake/modules/SearchInstalledSoftware.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/modules/SearchInstalledSoftware.cmake b/cmake/modules/SearchInstalledSoftware.cmake
index e1ca44973b..859b94108b 100644
--- a/cmake/modules/SearchInstalledSoftware.cmake
+++ b/cmake/modules/SearchInstalledSoftware.cmake
@@ -570,6 +570,9 @@ if(tmva-pymva)
   list(APPEND python_components NumPy)
 endif()
 find_package(Python3 3.8 COMPONENTS ${python_components})
+if(APPLE AND Python3_Development_FOUND)
+  set(Python3_LIBRARIES "-undefined dynamic_lookup")
+endif()
 
 #---Check for OpenGL installation-------------------------------------------------------
 if(opengl)
