From 143a5056b5f24d693670145a3af08c31ecf538dc Mon Sep 17 00:00:00 2001
From: Henry Schreiner <henryschreineriii@gmail.com>
Date: Mon, 27 Jan 2020 11:42:10 -0500
Subject: [PATCH] Add ability to remove flags

---
 CMakeLists.txt | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2c43dd579..3b11ffd48 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -393,12 +393,16 @@ if(GOOFIT_DEVICE STREQUAL CUDA)
         string(APPEND CMAKE_CUDA_FLAGS " -Xnvlink=--disable-warnings")
         string(APPEND CMAKE_CUDA_FLAGS " -Xcompiler=-Wno-attributes")
 
-        cmake_cuda_arch_select(FLAGS ARCH_FLAGS READABLE ARCH_FLAGS_readable ARCHS ${GOOFIT_ARCH})
-        string(REPLACE ";" ", " READ_ARCH_FLAGS "${ARCH_FLAGS_readable}")
-        string(REPLACE ";" " " ARCH_FLAGS "${ARCH_FLAGS}")
-        string(APPEND CMAKE_CUDA_FLAGS " ${ARCH_FLAGS}")
-        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS}" CACHE INTERNAL "")
-        message(STATUS "Compiling for GPU arch: ${READ_ARCH_FLAGS}")
+        if(NOT DEFINED GOOFIT_ARCH OR GOOFIT_ARCH)
+            cmake_cuda_arch_select(FLAGS ARCH_FLAGS READABLE ARCH_FLAGS_readable ARCHS ${GOOFIT_ARCH})
+            string(REPLACE ";" ", " READ_ARCH_FLAGS "${ARCH_FLAGS_readable}")
+            string(REPLACE ";" " " ARCH_FLAGS "${ARCH_FLAGS}")
+            string(APPEND CMAKE_CUDA_FLAGS " ${ARCH_FLAGS}")
+            set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS}" CACHE INTERNAL "")
+            message(STATUS "Compiling for GPU arch: ${READ_ARCH_FLAGS}")
+        else()
+            message(STATUS "Not compiling for a specific GPU arch")
+        endif()
 
         set(THRUST_INCLUDE_DIR "${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}")
 
