From 2494041502467ceee19f75f32bedf47899538e79 Mon Sep 17 00:00:00 2001
From: Chris Burr <chrisburr@users.noreply.github.com>
Date: Fri, 28 Jun 2019 14:52:18 +0200
Subject: [PATCH] Replace CXX_STANDARD with STANDARD_COMPILE_OPTION

---
 cmake/modules/CheckCompiler.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/modules/CheckCompiler.cmake b/cmake/modules/CheckCompiler.cmake
index dcf3c5aea07..2eb27b5611d 100644
--- a/cmake/modules/CheckCompiler.cmake
+++ b/cmake/modules/CheckCompiler.cmake
@@ -115,7 +115,7 @@ if(NOT CMAKE_CXX_STANDARD MATCHES "11|14|17")
 endif()
 
 # needed by roottest, to be removed once roottest is fixed
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++${CMAKE_CXX_STANDARD}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_CXX${CMAKE_CXX_STANDARD}_STANDARD_COMPILE_OPTION}")
 
 #---Check for libcxx option------------------------------------------------------------
 if(libcxx)
