diff --git a/cmake/modules/RootConfiguration.cmake b/cmake/modules/RootConfiguration.cmake
index c18d4d719d..8c367d4a35 100644
--- a/cmake/modules/RootConfiguration.cmake
+++ b/cmake/modules/RootConfiguration.cmake
@@ -651,6 +651,7 @@ string(REPLACE "c++11" "cxx11" features ${features}) # change the name of the c+
 set(configfeatures ${features})
 set(configargs ${ROOT_CONFIGARGS})
 set(configoptions ${ROOT_CONFIGARGS})
+set(configstd ${CMAKE_CXX${CMAKE_CXX_STANDARD}_STANDARD_COMPILE_OPTION})
 get_filename_component(altcc ${CMAKE_C_COMPILER} NAME)
 get_filename_component(altcxx ${CMAKE_CXX_COMPILER} NAME)
 get_filename_component(altf77 "${CMAKE_Fortran_COMPILER}" NAME)
diff --git a/config/root-config.in b/config/root-config.in
index 193551e601..115b510423 100755
--- a/config/root-config.in
+++ b/config/root-config.in
@@ -146,7 +146,7 @@ fi
 ### machine dependent settings ###

 # Set the C++ standard version
-cxxversionflag="-std=${cxxversion} "
+cxxversionflag="@configstd@ "

 case $arch in
 aix5)