diff --git a/bindings/pyroot/CMakeLists.txt b/bindings/pyroot/CMakeLists.txt
index 22d3ac8..1f42d06 100644
--- a/bindings/pyroot/CMakeLists.txt
+++ b/bindings/pyroot/CMakeLists.txt
@@ -16,8 +16,9 @@ endif()
 ROOT_STANDARD_LIBRARY_PACKAGE(PyROOT
                               NO_INSTALL_HEADERS
                               DICTIONARY_OPTIONS "-writeEmptyRootPCM"
-                              LIBRARIES Core Net Tree MathCore Rint ${PYTHON_LIBRARIES}
+                              LIBRARIES Core Net Tree MathCore Rint
                               DEPENDENCIES Core MathCore Net Rint Tree ROOTDataFrame ROOTVecOps)
+set_property(TARGET PyROOT APPEND PROPERTY LINK_FLAGS "-undefined dynamic_lookup") # macOS Only
 ROOT_LINKER_LIBRARY(JupyROOT ../JupyROOT/src/*.cxx DEPENDENCIES Core CMAKENOEXPORT)

 if(MSVC)
diff --git a/tmva/pymva/CMakeLists.txt b/tmva/pymva/CMakeLists.txt
index 2e786ed..317c8f5 100644
--- a/tmva/pymva/CMakeLists.txt
+++ b/tmva/pymva/CMakeLists.txt
@@ -19,7 +19,7 @@ set(PY_HEADERS ${CMAKE_SOURCE_DIR}/tmva/pymva/inc/TMVA/PyMethodBase.h

 ROOT_STANDARD_LIBRARY_PACKAGE(PyMVA
                               HEADERS ${PY_HEADERS}
-                              LIBRARIES ${PYTHON_LIBRARIES}
                               DEPENDENCIES Core Matrix Thread RIO TMVA)
+set_property(TARGET PyMVA APPEND PROPERTY LINK_FLAGS "-undefined dynamic_lookup")

 ROOT_ADD_TEST_SUBDIRECTORY(test)
