From 284359eb5658ac9dda8a73af105ec07c87f27825 Mon Sep 17 00:00:00 2001
From: Stefan Wunsch <stefan.wunsch@cern.ch>
Date: Wed, 8 Jul 2020 10:10:24 +0200
Subject: [PATCH] [PyROOT] Install additional PyROOT headers

---
 bindings/pyroot/pythonizations/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/bindings/pyroot/pythonizations/CMakeLists.txt b/bindings/pyroot/pythonizations/CMakeLists.txt
index ec9f86a974b..0b3b6983e19 100644
--- a/bindings/pyroot/pythonizations/CMakeLists.txt
+++ b/bindings/pyroot/pythonizations/CMakeLists.txt
@@ -16,6 +16,8 @@ if(dataframe)
     list(APPEND PYROOT_EXTRA_SOURCE
         src/RDataFramePyz.cxx
         src/RTensorPyz.cxx)
+    list(APPEND PYROOT_EXTRA_HEADERS
+        inc/RNumpyDS.hxx)
 endif()
 
 set(py_sources
@@ -152,4 +154,7 @@ install(DIRECTORY ${localruntimedir}/ROOT
         DESTINATION ${CMAKE_INSTALL_LIBDIR}
         COMPONENT libraries)
 
+# Install headers required by pythonizations
+install(FILES ${PYROOT_EXTRA_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/ROOT)
+
 ROOT_ADD_TEST_SUBDIRECTORY(test)
